### PR TITLE
Start to refactor and document `rt.backtrace.dwarf` 

### PR DESCRIPTION
**Pull request by Geod24**
I started to work on https://issues.dlang.org/show_bug.cgi?id=20510 and realized that the code in this module is quite old, and inefficient. I have a few refactorings coming up, but before that, I wanted to submit the non-controversial commits.

### CODE REVIEW

File: src/rt/backtrace/dwarf.d [hunk 1]
```diff
@@ -12,7 +49,7 @@
 
 module rt.backtrace.dwarf;
 
-private import core.internal.execinfo;
```
Comment (MoonlightSentinel) [inline src/rt/backtrace/dwarf.d:15]: Why?

File: src/rt/backtrace/dwarf.d [hunk 2]
```diff
@@ -206,7 +260,42 @@ void resolveAddresses(const(ubyte)[] debugLineSectionData, Location[] locations,
...
+ * Run the state machine to generate line number matrix
+ *
+ * Line number informations generated by the compiler are stored in the
+ * `.debug_line` section. Conceptually, they can be seen as a large matrix,
+ * with row such as "file", "line", "column", "is_statement", etc...
+ * However such a matrix would be too big to store in an object file,
+ * so DWARF instead generate this matrix using bytecode fed to a state machine.
+ *
+ * Note:
+ * Each compilation unit (module in D) ccan have its own line number program.
```
Comment (kinke) [inline src/rt/backtrace/dwarf.d:287]: Little typo ('ccan'); and a compilation unit isn't necessarily a D module.

Comment (Geod24) [reply src/rt/backtrace/dwarf.d:287]: Thanks, fixed

**Comment by thewilsonator** [comment]
@dlang-bot quit slacking!

**Pull request merged**